### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by TekyRey**
### **Math Magicians: tests**

In this project we:

- Wrote unit tests for the files operate.js and calculate.js using Jest.
- Used Jest snapshots to test the header, navbar, home, quotes, and calculator components.
- Used React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      total: 0,
+      next: null,
+      operation: null,
+    }, 'AC');
+    expect(result).toEqual({
+      total: null,
+      next: null,
+      operation: null,
+    });
+  });
```
Comment (DammyShittu) [inline src/tests/calculate.test.js]: - [ ] Kindly add more tests covering most scenarios present in the calculate.js file.

File: src/tests/component.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  });
+
+  it('Should render the navbar component', () => {
+    <Router>
+      const navbar = renderer.create(
+      <Navbar />
+      ).toJSON(); expect(navbar).toMatchSnapshot();
+    </Router>;
+  });
+});
```
Comment (DammyShittu) [inline src/tests/component.test.js:33]: - [ ] Kindly add more tests simulating user interaction using the React Testing Library. These tests should check for firing of events like button clicks and resulting calculations.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @TekyRey @kendoriddy,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

**Hi @Team,** 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**